modules/mobile/app/controllers/mobile/v0/facilities_info_controller.rb

```diff
@@ -39,13 +39,18 @@ def sort(facilities, sort_method)
 
       def sort_by_recent_appointment(facilities)
         facilities.sort_by(&:name) # entries not in recent appointments will be sorted alphabetically
-        appointments = Mobile::V0::Appointment.get_cached(@current_user).sort_by(&:start_date_utc)
+        appointments = Mobile::V0::Appointment.get_cached(@current_user)&.sort_by(&:start_date_utc)
         if appointments.blank?
           raise Common::Exceptions::RecordNotFound.new(
             detail: 'Could not fetch user appointments', source: self.class.to_s
```

Comment:
TBH, the existing code seems slightly wrong. I'm starting to see what Andrew was talking about with this bug occurring because we're using the API and not the mobile app. This code should work through the mobile app because we pre-cache appointments on login. And we intend this code to be used as part of the appointments flow and it might not be possible to get to this request in the mobile app without first navigating to the appointments index, which will also cache appointments. But we've already added a feature flag to the login pre-cache because we know it might cause performance issues and may need to be turned off. So that's fragile. And if we make a design change that allows the user to navigate to direct schedule without hitting the index first, this becomes broken. 

So yes it would take two points of failure for this to be a problem, but I also think this could be easily fixed by extracting the `fetch_cached_or_service` from the appointments controller into a service object or model method and using it here. So instead of fetching cached here, we just do a cached or fetch. What do you guys think @aherzberg @jperk51 ?

At the very least, I think we should reword this error message to something like, "User does not have cached appointments". But it's also worth pointing out that this might not indicate an error at all. It could just be a user who has no previous appointments, in which case it would probably make sense to just return an empty response.

Reply:
Jayson and I discussed this a bit when implementing it. I think the idea around just using the cache was that doing an appointment fetch on top of the original request was going to really draw out the response time since appointments is pretty expensive. we were going to use the logs to see how often there wasn't a cache and make an informed decision on how to address it. I personally don't mind putting a fetch in there but It definitely complicates things when you've got appointment fetching happening outside the appointment controller. 

I do agree we can update the wordage of the raised error to say its the cache that's empty.

Reply:
I'm not sure I understand what you mean about fetching outside the appointments controller complicating things. We already do it in the users controller during the pre-cache. 

I can see the argument that a controlled error is better than a timeout. However, we're treating this as a 404, which no longer seems correct. I think this is more like a 422. Also, I think we should consider the possibility that a user might not have any prior appointments, in which case perhaps an empty response might be better than an error.

Reply:
I too was wondering about the event the user legitimately doesn't have any appointments. I wasn't sure how front end handles our errors, so I suppose I thought they would provide a message to the user if they don't have appointments.

I see this PR was approved, but I'm hesitant to move forward with it without a conclusion being made on if this should be tweaked. Would the proposed changes be a separate ticket, or would it be handled in this existing PR?

Reply:
The person who approved is not a mobile team developer. We just don't have code owner approval set up correctly to block merging until you have approval from a mobile team developer if the code is in the mobile module. 

I think we need to come to a consensus about what we want to do before we merge this. I'll hit you up on slack.

Reply:
Ok just updated with the logic discussed via slack and added new specs. Ready for review again